### ISSUE TITLE
[FEATURE REQUEST] Add Permanent Links for Friends that Own, Achievements, and Cards

### ISSUE THREAD

**Issue by RathaWynter** [IsThereAnyDeal/AugmentedSteam#2085]
### Current Behavior (if any)
Many Steam store pages do not always display the same consistent information.
1: Many store pages such as for "Free" games do not have links to 'friends that want / own / play' page. (See attached images.)

2: Steam store pages which have less than 3-4 achievements do not have a link to the achievements page. (See attached images.)

3: Some Steam games have cards, but their store pages do not link you to their 'card page.' (Only example i could easily find via googling was an adult game and not sure if that'd be appropriate to post here. Will include if needed / attempt to find an alternative.)

### Desired Behavior
It would be nice if Augmented Steam could add permanent links to each of these elements on ALL store pages, to replace this inconsistent Steam behavior that often hides page elements under certain conditions.

### Motivation / Use Case for Changing the Behavior
1: This store page is a free-to-claim game. However there is no link to friends that own it on the store page unlike what would be seen on a standard paid-game store page.
https://store.steampowered.com/app/3056590/Swap_Fire/

Link to the hidden 'friends that own page.'
https://steamcommunity.com/id/[STEAM_COMMUNITY_NAME]/friendsthatplay/553850

![1_Steam_Friends_That_Play_Comparison](https://github.com/user-attachments/assets/5b1c0dd6-bce6-407c-aa8a-d794ed48588b)

-

2: This store page only has 2 achievements with no link to the achievements page.
https://store.steampowered.com/app/210970/The_Witness/

Achievements page with additional details / links / hints / descriptions / stats / not viewable or linked from the store page.
https://steamcommunity.com/stats/210970/achievements

![2_Steam_Achievements_Comparison](https://github.com/user-attachments/assets/cbaffa7b-c064-481d-8371-ee226669abb1)

**Comment by candela97** [comment]
For the cards issue, the option "Show badge progress" under App page will show up with a link to the cards you own if the game has cards. Does that do what you want?

**Comment by RathaWynter** [comment]
> For the cards issue, the option "Show badge progress" under App page will show up with a link to the cards you own if the game has cards. Does that do what you want?

Thanks for the reply. It does do what I'm looking for most of the time, and now I'm having a hard time finding an example of a game that Augmented Steam does not cover properly at the moment. Have been using ES and AS for years and there used to be a bunch of cases where it didn't show the widget when it should have. (Maybe those issues are fixed now.)

The game that I'd mentioned in the original post also seems to be covered as well despite having double checked it the night before while writing the post, maybe that was a temporary issue. Should perhaps strike the 'badges / card page link' from the feature request for the time being until I can find a reliable example of it not displaying the widget for games which do actually have cards.

Some edge cases that i can think of to check in the future (for my own reference since Im thinking about it);
- Games which dont list trading cards, but do have card drops.
- Games which dont list trading cards, and dont have drops, but do have gem-craftable card sets.
- Games which list trading cards on the store page, but dont actually have trading cards.

**Comment by candela97** [comment]
Yeah the method we use to determine if a game has cards has been updated, and is now the same as the one Steam uses to display owned cards on https://steamcommunity.com/my/games. So if it's not showing up when it should it's likely a problem with Steam.

For links to "friends that want" etc., can you open `https://store.steampowered.com/api/appuserdetails?appids=3056590` and see if there's anything under the `data.friendswant` or `data.friendsown` fields?

**Comment by candela97** [comment]
Oh! I missed the "Steam is learning about this game" tag (they should really make this more noticable), but that could very well be the cause.

Can you try `https://api.steampowered.com/IPlayerService/GetFriendsGameplayInfo/v1/?appid=3056590&access_token=` and see if it makes a difference? You can grab your access token by inspecting Steam web API requests or here `https://store.steampowered.com/pointssummary/ajaxgetasyncconfig`.

**Comment by RathaWynter** [comment]
Using that method does appear to return ownership results for the account.
![Area - 2024-12-03 - 18 15 48](https://github.com/user-attachments/assets/f55395ed-0552-4221-8027-5e1b7e7a8868)

**Comment by candela97** [comment]
We're getting somewhere then. It'll be a bit of a hassle to resolve the steamids, display avatars and attach miniprofile hovers just like Steam does though 😅
